### PR TITLE
requestNewAuthToken will clear token if it fails

### PR DESCRIPTION
**Pull request by bill1600**
Function requestNewAuthToken, in auth_token.c, will clear the result token if it fails.

### CODE REVIEW

File: src/auth_token.c [hunk 1]
```diff
@@ -66,6 +66,7 @@ int requestNewAuthToken(char *newToken, size_t len, int r_count)
 
 	struct token_data data;
 	data.size = 0;
+	newToken[0] = '\0';
```
Comment (shilpa24balaji) [inline src/auth_token.c]: I feel memset would be better for the input argument len

Comment (bill1600) [reply src/auth_token.c]: OK.

**Comment by shilpa24balaji** [comment]
Fixes #301

**Comment by shilpa24balaji** [comment]
@sadhyama please review this.

**Pull request merged**